### ISSUE TITLE
pairing_key empty

### ISSUE THREAD

**Issue by stianale** [chaidiscovery/chai-lab#128]
Hi,

The code for specifying pairing_key for generating the pqt files is not working unless the taxa in the headers in the input a3m file are denoted "OS=" and you specify uniprot_hits.a3m as the name of the file. This seems strange, as the get_tax_name function in the species.py script seems reasonable.

By the way, the input file must be named <sequence_database>_hits.a3m and not hits_<sequence_database>.a3m as stated in the helper function of the script converting a3m to pqt.

**Comment by wukevin** [comment]
I agree that this behavior with parsing only UniProt hits' organism information appears strange on the surface. This logic actually follows the procedure in AlphaFold 3 and AlphaFold 2 multimer, where "The UniProt search result is kept separate and used to provide cross-chain genetic information" (see [AF3 supplement](https://static-content.springer.com/esm/art%3A10.1038%2Fs41586-024-07487-w/MediaObjects/41586_2024_7487_MOESM1_ESM.pdf)). As such, the "out of the box" logic is specific to uniprot style headers and uniprot files.

More broadly, our provided code for generating pqt files is mainly there to serve as a reference, and your specific use case may vary. We hope that it's easy to build additional parsers, and let use know if you need some guidance on that front. 

Regarding the naming of a3m files, can you share the directory structure where you're running into this issue? From my own testing, both `db_hits.a3m` and `hits_db.a3m` seem to work correctly: https://github.com/chaidiscovery/chai-lab/blob/a25da2517186ae6739a155f3b8a87e0efe5cf815/chai_lab/data/parsing/msas/aligned_pqt.py#L212-L216

We're updating our documentation to better address the points you've raised; see https://github.com/chaidiscovery/chai-lab/pull/129.

Thanks for the feedback!

**Comment by stianale** [comment]
> During training, Chai-1 is trained by using only UniProt hits for pairing information across MSAs for different chains, hence this default behavior. For more information refer to these sources: [(1)](https://www.biorxiv.org/content/10.1101/2024.10.10.615955v2.full) and [(2)](https://static-content.springer.com/esm/art%3A10.1038%2Fs41586-024-07487-w/MediaObjects/41586_2024_7487_MOESM1_ESM.pdf). We expect that this logic will produce good results in general for users, but we hope it's easy to build additional parsers that can specify different modes of pairing information; let use know if you need some guidance on that front.
> 
> Regarding the naming of a3m files, can you share the directory structure where you're running into this issue? From my own testing, both `db_hits.a3m` and `hits_db.a3m` seem to work correctly:
> 
> https://github.com/chaidiscovery/chai-lab/blob/a25da2517186ae6739a155f3b8a87e0efe5cf815/chai_lab/data/parsing/msas/aligned_pqt.py#L212-L216
> 
> We're updating our documentation to better address the points you've raised; see #129.
> 
> Thanks for the feedback!

The naming of the input a3m files works for me now after cloning by the latest repo commit. The problem with files named hits_source_database.a3m persisted for me with a version of your repo (kevin/msa-context) with sha "d4e7cb8fc3303f072946f450e8977f8eceeefb15".

I also got pairing_key working for uniref90 after changing the line "src == MSADataSource.UNIPROT" to "src in {MSADataSource.UNIPROT, MSADataSource.UNIREF90}" in the function merge_multi_a3m_to_aligned_dataframe(), as well as changing "TaxID=(\d+)" to "Tax=([^\s]+ [^\s]+)" in species.py for the uniref90 condition. Thanks for the quick and helpful response!
